frontend/portals/MaintenanceMode/index.jsx

```diff
@@ -85,63 +91,103 @@ class MaintenanceMode extends Component {
   };
 
   /**
+   * Checks Dates.
+   * @returns {boolean}
+   */
+  checkDate = () => {
+    const parseStartDate = startDate.replaceAll('/', '-').replace(' - ', 'T');
+    const parseEndDate = endDate.replaceAll('/', '-').replace(' - ', 'T');
+
+    const now = new Date();
+
+    if (!startDate && !endDate) {
+      // no times provide. so its always valid
+      return true;
+    }
+
+    if (!startDate) {
+      // no start date given. only check valid end date
+      return new Date(parseEndDate) > now;
+    }
+
+    if (!endDate) {
+      // no end date given. only check valid start date
+      return new Date(parseStartDate) < now;
+    }
+
+    return new Date(parseStartDate) < now && new Date(parseEndDate) > now;
+  };
+
+   /**
+   * Checks if there is a page whitelist and only enables maintenance for these pages.
+   * @param {Object} currentRoute App version
+   * @returns {boolean}
+   */
+   hasPageWhiteList = currentRoute => maintenancePagesWhitelist.includes(currentRoute.pattern) ||
```

Comment:
`hasPageWhiteList` = true, when the whitelist is empty

I think the variable should have an other name?

Reply:
renamed